src/utils/buffer.ts

```diff
@@ -12,3 +14,134 @@ export const hexToBuffer = (value: string): Buffer => {
   }
   return Buffer.from(value, 'hex');
 };
+
+/**
+ * Unpacks a buffer size
+ *
+ * @param {number} n The size of the buffer to unpack
+ * @param {Buffer} buff The buffer to unpack
+ *
+ * @return {[Buffer, Buffer]} The unpacked buffer followed by the rest of the buffer
+ */
+export const unpackLen = (n: number, buff: Buffer): [Buffer, Buffer] => {
+  return [
+    buff.slice(0, n),
+    buff.slice(n)
+  ];
+};
+
+/**
+ * Unpacks an integer from a buffer
+ *
+ * @param {number} n The size of the number in bytes
+ * @param {boolean} signed If the number is signed
+ * @param {Buffer} buff The buffer to unpack
+ *
+ * @return {[number, Buffer]} The unpacked number followed by the rest of the buffer
+ */
+export const unpackToInt = (n: number, signed: boolean, buff: Buffer): [number, Buffer] => {
+  let retInt;
+  const slicedBuff = buff.slice(0, n);
+  if (n === 1) {
+    if (signed) {
+      retInt = slicedBuff.readInt8();
+    } else {
+      retInt = slicedBuff.readUInt8();
+    }
+  } else if (n === 2) {
+    if (signed) {
+      retInt = slicedBuff.readInt16BE();
+    } else {
+      retInt = slicedBuff.readUInt16BE();
+    }
+  } else if (n === 4) {
+    if (signed) {
+      retInt = slicedBuff.readInt32BE();
+    } else {
+      retInt = slicedBuff.readUInt32BE();
+    }
+  } else if (n === 8) {
+    // We have only signed ints here
+    if (parseInt(nodeMajorVersion) > 8) {
+      retInt = Number(slicedBuff.readBigInt64BE());
+    } else {
+      retInt = slicedBuff.readIntBE(0, 8);
+    }
+
+  }
+
+  return [
+    retInt,
+    buff.slice(n)
+  ];
+};
+
+/**
+ * Unpacks a float from a buffer
+ *
+ * @param {Buffer} buff The buffer to unpack
+ *
+ * @return {[number, Buffer]} The unpacked float followed by the rest of the buffer
+ */
+export const unpackToFloat = (buff: Buffer): [number, Buffer] => {
+  const n = 8;
+  const retFloat = buff.slice(0, n).readDoubleBE();
+  return [
+    retFloat,
+    buff.slice(n)
+  ];
+};
+
+/**
+ * Unpacks a hex from a buffer
+ *
+ * @param {number} n The size of the hex to unpack
+ * @param {Buffer} buff The buffer to unpack
+ *
+ * @return {[string, Buffer]} The unpacked hex followed by the rest of the buffer
+ */
+export const unpackToHex = (n: number, buff: Buffer): [string, Buffer] => {
+  const unpackedRet = unpackLen(n, buff);
+  const unpackedHex = bufferToHex(unpackedRet[0]);
+
+  return [
+    unpackedHex,
+    unpackedRet[1]
+  ];
+};
+
+/**
+ * Transforms buffer to hex
+ *
+ * @param {Buffer} buff The buffer to be transformed to hex
+ *
+ * @return {string} Hexadecimal of the buffer
+ */
+export const bufferToHex = (buff: Buffer): string => {
+  return buff.toString('hex');
```

Comment:
I agree it's simple to call `toString('hex')` but after a long time without using buffer in js I had to search how to transform buffer to hex. So even though is a simple method I think it can be useful for a more direct use.

Is there any downside of having it? I don't see any, that's why I prefer having it.

Reply:
Ok. I was just wondering whether it was really useful.